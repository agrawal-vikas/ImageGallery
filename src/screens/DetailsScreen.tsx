import React from 'react';
import { View, Text, Image, StyleSheet, Share, ScrollView } from 'react-native';
import { useRoute } from '@react-navigation/native';
import RenderHtml from 'react-native-render-html';
import { DetailsScreenRouteProp } from './types/types';
import Tag from '../components/Tag';
import { extractDescriptionWithoutImage, extractNameAndEmail, formatUtcToReadableDate } from '../utils';

const DetailsScreen = () => {
    const route = useRoute<DetailsScreenRouteProp>();
    const image = route.params.image;
    return (
        <View style={styles.container}>
            <Image
                source={{ uri: image.media.m }}
                style={styles.image}
                resizeMode="contain"
                testID="image"
            />
            <View style={styles.metadata}>
                <View style={styles.row}>
                    <Text style={styles.label}>Author:</Text>
                    <Text style={styles.value}>{extractNameAndEmail(image.author)?.name}</Text>
                </View>
                <View style={styles.row}>
                    <Text style={styles.label}>Published:</Text>
                    <Text style={styles.value}>{formatUtcToReadableDate(image.published)}</Text>
                </View>
                <View style={styles.row}>
                    <Text style={styles.label}>Description:</Text>
                    <RenderHtml
                        contentWidth={styles.metadata.padding || 0}
                        source={{ html: extractDescriptionWithoutImage(image.description) || '' }}
                    />
                </View>
                <View style={styles.row}>
                    <Text style={styles.label}>Tags:</Text>
                    <View style={{ ...styles.tags, ...styles.value }}>
                        {image.tags.split(' ').map((tag, index) => (
                            <Tag key={index} tag={tag} />
                        ))}
                    </View>
                </View>
            </View>
        </View >
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
    },
    image: {
        flex: 1,
        width: '100%',
    },
    metadata: {
        padding: 32,
        backgroundColor: '#f8f8f8',

    },
    author: {
        fontSize: 16,
        color: '#333',
    },
    row: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        marginBottom: 8,
    },
    label: {
        fontSize: 16,
        fontWeight: 'bold',
        color: '#333',
    },
    value: {
        fontSize: 16,
        color: '#333',
    },
    tags: {
        flexDirection: 'row',
        flexWrap: 'wrap',
        gap: 4
    }
});

export default DetailsScreen;