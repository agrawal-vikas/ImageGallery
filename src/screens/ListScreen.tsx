import React from 'react';
import { useNavigation, NavigationProp } from '@react-navigation/native';
import { ScrollView, View, Text, Image, StyleSheet } from 'react-native';
import { searchPhotos } from '../services/services';
import { RootStackParamList, TypeImage } from './types/types';
import ImageSection from '../components/ImageSection';
import ErrorBoundary from '../components/ErrorBoundry';

const ListScreen = () => {
    const navigation = useNavigation<NavigationProp<RootStackParamList>>();

    const handleImagePress = (image: TypeImage) => {
        navigation.navigate('Details', { image });
    };

    return (
        <ScrollView style={styles.container}>
            <ImageSection
                title="Kittens"
                tag="kittens"
                onImagePress={handleImagePress} />
            <ImageSection
                title="Dogs"
                tag="dogs"
                onImagePress={handleImagePress} />
            <ImageSection
                title="Public"
                tag="public"
                onImagePress={handleImagePress} />
        </ScrollView>
    );
}
const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        padding: 10,
    }
});

export default ListScreen;