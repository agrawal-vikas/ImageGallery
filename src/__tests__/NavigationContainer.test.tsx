import React from 'react';
import { render } from '@testing-library/react-native';
import NavigationContainerComponent, { RootStack } from '../NavigationContainer';
import { NavigationContainer } from '@react-navigation/native';
import ListScreen from '../screens/ListScreen';
import DetailsScreen from '../screens/DetailsScreen';

jest.mock('../screens/ListScreen', () => jest.fn(() => null));
jest.mock('../screens/DetailsScreen', () => jest.fn(() => null));

jest.mock('@react-navigation/native-stack', () => {
    const { Text } = require('react-native');
    return {
        createNativeStackNavigator: jest.fn(() => ({
            Navigator: ({ children }: { children: React.ReactNode }) => <>{children}</>,
            Screen: ({ children }: { children: React.ReactNode }) => <>{children}</>,
            config: {
                initialRouteName: 'List',
                screens: {
                    List: () => <Text>List</Text>,
                    Details: () => <Text>Details</Text>,
                }
            },
        })),
    };
});

describe('NavigationContainer', () => {
    it('renders the navigation container without crashing', () => {
        const { toJSON } = render(
            <NavigationContainerComponent />
        );
        expect(toJSON()).toMatchSnapshot();
    });

    it('has the correct initial route', () => {
        expect(RootStack.config.initialRouteName).toBe('List');
    });

    it('renders the ListScreen component for the "List" route', () => {
        const { toJSON } = render(
            <NavigationContainer>
                <RootStack.Navigator>
                    <RootStack.Screen name="List" component={ListScreen} />
                </RootStack.Navigator>
            </NavigationContainer>
        );
        expect(toJSON()).toMatchSnapshot();
    });

    it('renders the DetailsScreen component for the "Details" route', () => {
        const { toJSON } = render(
            <NavigationContainer>
                <RootStack.Navigator>
                    <RootStack.Screen name="Details" component={DetailsScreen} />
                </RootStack.Navigator>
            </NavigationContainer>
        );
        expect(toJSON()).toMatchSnapshot();
    });
});