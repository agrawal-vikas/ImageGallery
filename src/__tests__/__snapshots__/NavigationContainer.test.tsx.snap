// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationContainer renders the DetailsScreen component for the "Details" route 1`] = `null`;

exports[`NavigationContainer renders the ListScreen component for the "List" route 1`] = `null`;

exports[`NavigationContainer renders the navigation container without crashing 1`] = `null`;
